diff_drive_controller/src/diff_drive_controller.cpp

```diff
@@ -101,15 +101,6 @@ controller_interface::return_type DiffDriveController::update_reference_from_sub
   const rclcpp::Time & time, const rclcpp::Duration & /*period*/)
 {
   auto logger = get_node()->get_logger();
-  if (get_lifecycle_state().id() == State::PRIMARY_STATE_INACTIVE)
-  {
-    if (!is_halted)
```

Comment:
Do we also need to remove the is_halted variable, if it is a member variable?

Reply:
we use it in on_deactivate() transition

Reply:
Can't we simply call halt in the deactivate method always, instead of using this is_halted variable?

What do you think?

I can recheck the code later and get back to you

Reply:
I think we could, sure.